flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphContext.java

```diff
@@ -76,12 +76,13 @@ public interface StreamGraphContext {
     boolean modifyStreamNode(List<StreamNodeUpdateRequestInfo> requestInfos);
 
     /**
-     * Check whether all upstream nodes of the stream node have finished executing.
+     * Check whether the upstream nodes of the stream node with pointed type number have finished
+     * executing.
      *
      * @param streamNode the stream node that needs to be determined.
      * @return true if all upstream nodes are finished, false otherwise.
      */
-    boolean areAllUpstreamNodesFinished(ImmutableStreamNode streamNode);
+    boolean checkUpstreamNodesFinished(ImmutableStreamNode streamNode, Integer typeNumber);
```

Comment:
@Nullable, and some additional comments are required to clarify the meaning of a null value for typeNumber.